internal/mode/static/telemetry/collector_test.go

```diff
@@ -348,17 +390,184 @@ var _ = Describe("Collector", Ordered, func() {
 					fakeConfigurationGetter.GetLatestConfigurationReturns(&dataplane.Configuration{})
 				})
 				It("should error on nil latest graph", func() {
+					expectedError := errors.New("latest graph cannot be nil")
 					fakeGraphGetter.GetLatestGraphReturns(nil)
 
 					_, err := dataCollector.Collect(ctx)
-					Expect(err).To(HaveOccurred())
+					Expect(err).To(MatchError(expectedError))
 				})
 
 				It("should error on nil latest configuration", func() {
+					expectedError := errors.New("latest configuration cannot be nil")
 					fakeConfigurationGetter.GetLatestConfigurationReturns(nil)
 
 					_, err := dataCollector.Collect(ctx)
-					Expect(err).To(HaveOccurred())
+					Expect(err).To(MatchError(expectedError))
+				})
+			})
+		})
+	})
+
+	Describe("NGF replica count collector", func() {
+		When("collecting NGF replica count", func() {
+			When("it encounters an error while collecting data", func() {
+				It("should error if the kubernetes client errored when getting the Pod", func() {
+					expectedErr := errors.New("there was an error getting the Pod")
+					k8sClientReader.GetCalls(
```

Comment:
I've been wrong about this before, but I think we can use `GetReturns` here since we just care about the error, and the Pod Get call is the first Get Call.

Reply:
I had used GetReturnsOnCall (maybe `GetReturns` I forgot), however decided after @pleshakov 's comment about

> The approach with specifying the order here is a bit brittle.
What if we add some other collector that also call Gets and change the order of get calls?

that if there was another function in `Collect()` which used the `Get` function and for some reason we swapped the orders so that `Get` function was called prior to this one, this would no longer work. 

`GetReturns` works for now, but I think would not be the best approach moving forward. Also, for all other testing `Get` calls, we would need to use the `GetCalls` so it doesn't make the most sense to make this an edge case and use `GetReturns` just because it currently is the first `Get` call.

Reply:
I see. This makes sense.

I wonder if testing all the collection functions through the `Collect` method is best. It would probably be easier to test each collection function directly with `go test`. I'm concerned that the GetCall stub is going to become unwieldy as we add more collection functions.

Reply:
yep, I share you concern.

However, the benefit of testing through Collect is that allows us to change the design of those functions.

Additionally, If we test individual functions too, we still need to:
- Collect - happy path of all functions.
- Collect - unhappy path of each function (to make sure error is propagated to collect).

Which will still require us to configure APIReader mocks, unless we change the design of the Collector and introduce injection for each collector functions.